### DISCUSSION TITLE
Example of createFileSessionStorage usage needed

### DISCUSSION THREAD

**Discussion by waldeku** [remix-run/remix#8735] [Q&A]
Hi. 
I create a project where I need to store more then 4kb of data in cookie. Due to this reason I choose the file-backed sessions approach. 
Following the [documentation ](https://remix.run/docs/en/main/utils/sessions#createfilesessionstorage-node) I use `createFileSessionStorage` however I have a feeling that there is something wrong with my setup. 
As I understood once the `createFileSessionStorage` is run is should create a file in `/app/sessions` folder but in my case it does not happen. 
Can anyone share with my a very basic example of how to use a file-backed sessions approach?
Furthermore, we could add it later to official Remix examples.

**Comment by kiliman** [comment]
The example `/app/sessions` was meant as a placeholder. On typical systems, that would try to create a folder in the _root_ of your file system that the web app would not have permission for.

To create a `sessions` folder **relative** to your _project_ directory, remove the leading `/`.

```ts
const { getSession, commitSession, destroySession } = createFileSessionStorage({
  // The root directory where you want to store the files.
  // Make sure it's writable!
  dir: 'sessions',
  cookie: sessionCookie,
});
```
Consider submitting a PR to update the documentation to use the above code. This would clarify things for future developers.

I created a StackBlitz example. https://stackblitz.com/edit/remix-run-remix-dfi3rb

<img width="218" alt="image" src="https://github.com/remix-run/remix/assets/47168/747a3aa0-5f5c-409f-8b59-00bd6697a90b">
